### ISSUE TITLE
Segmentation fault when running tests

### ISSUE THREAD

**Issue by aralroca** [oven-sh/bun#11664]
### How can we reproduce the crash?

I have 2000 tests, some of them when updating Bun to older versions, some of them started to fail (19), now with the new version 1.1.13, running my tests gets a segmentation fault. To reproduce it, creating many tests with happy-dom and using/manipulating files is necessary.

### Relevant log output

```shell
Bun Canary v1.1.13-canary.1 (8d6f1951) macOS Silicon
Args: "bun" "test"
Features: jsc(2) bunfig define(16) external(2) macros(15) spawn(2) transpiler_cache(9) tsconfig_paths(2) tsconfig(9) 
Builtins: "node:buffer" "node:child_process" "node:crypto" "node:fs" "node:fs/promises" "node:http" "node:https" "node:net" "node:path" "node:stream" "node:stream/web" "node:string_decoder" "node:url" "node:util/types" "node:vm" "node:zlib" 
Elapsed: 1226ms | User: 1112ms | Sys: 291ms
RSS: 0.22GB | Peak: 0.22GB | Commit: 0.44GB | Faults: 75
```

### Stack Trace (bun.report)

Bun v1.1.13-canary ([`8d6f195`](<https://github.com/oven-sh/bun/tree/8d6f19516f7f9932e9b52695998f0f06ccad2e27>)) on macos aarch64 [TestCommand]

Segmentation fault at address 0x00000018

- *2 unknown/js code*
- `JSC::JSModuleNamespaceObject::overrideExportValue`
- `JSMock__jsModuleMock`
- *1 unknown/js code*
- `llint_entry`
- *1 unknown/js code*
- `llint_entry`
- `vmEntryToJavaScript`
- `JSC::Interpreter::executeCall`

Features: jsc, bunfig, define, external, macros, spawn, transpiler\_cache, tsconfig\_paths, tsconfig

**Comment by Jarred-Sumner** [comment]
Is there any code you could share to reproduce? As far as I can tell, this is something related to mocking functions or using spyOn. But it's hard to say what/where exactly.

**Comment by aralroca** [comment]
Yes, invited so that you have direct access to the https://github.com/brisa-build/brisa repo, just run `bun test` to reproduce it

**Comment by Jarred-Sumner** [comment]
debug assertion failure:
```zig
============================================================
Bun v1.1.13 (25a09d88) macOS Silicon
Args: "/Users/jarred/Code/bun/build/bun-debug" "test"
Features: jsc(2) bunfig define(2) external(2) macros(11) spawn(2) transpiler_cache(5) tsconfig_paths(4) tsconfig(11) 
Builtins: "node:buffer" "node:child_process" "node:crypto" "node:fs" "node:fs/promises" "node:http" "node:https" "node:net" "node:path" "node:stream" "node:stream/web" "node:string_decoder" "node:url" "node:util/types" "node:vm" "node:zlib"
Elapsed: 4546ms | User: 8304ms | Sys: 252ms
RSS: 0.28GB | Peak: 0.28GB | Commit: 0.66GB | Faults: 95

panic(main thread): reached unreachable code
/Users/jarred/Code/bun/src/bun.zig:3273:43: 0x10000b59f in assert (bun-zig)
        if (comptime Environment.isDebug) unreachable;
                                          ^
/Users/jarred/Code/bun/src/router.zig:470:47: 0x100a94233 in load__anon_139467 (bun-zig)
                                    bun.assert(bun.path.isSepAny(entry.dir.ptr[base_dir.len - 1]));
                                              ^
/Users/jarred/Code/bun/src/router.zig:357:18: 0x10055e5c3 in loadAll__anon_108116 (bun-zig)
        this.load(ResolverType, resolver, root_dir_info, base_dir);
                 ^
/Users/jarred/Code/bun/src/router.zig:507:38: 0x10009162f in loadRoutes__anon_55599 (bun-zig)
    this.routes = RouteLoader.loadAll(this.allocator, this.config, log, ResolverType, resolver, root_dir_info, base_dir);
                                     ^
/Users/jarred/Code/bun/src/bun.js/api/filesystem_router.zig:169:26: 0x100ef990b in constructor (bun-zig)
        router.loadRoutes(&log, root_dir_info, Resolver, &vm.bundler.resolver, router.config.dir) catch {
                         ^
codegen/ZigGeneratedClasses.cpp:19831:17: 0x10312ad37 in construct (/Users/jarred/Code/bun/build/codegen/ZigGeneratedClasses.cpp)
???:?:?: 0x11a25448f in ??? (???)
???:?:?: 0x105cccd5f in _llint_entry (???)
???:?:?: 0x11a297edb in ??? (???)
???:?:?: 0x11a29131b in ??? (???)

Stop reason: EXC_BREAKPOINT (code=1, subcode=0x10000ab00)
Child list expansion has timed out.

```

**Comment by paperdave** [comment]
@aralroca can you re-invite me to that repository?

**Comment by aralroca** [comment]
> @aralroca can you re-invite me to that repository?

Done!

**Comment by paperdave** [comment]
<img width="810" alt="image" src="https://github.com/oven-sh/bun/assets/24465214/9d5ebb2d-e9e0-4b2d-8c7c-33f21832fbd7">
consistant reproduction, thanks

**Comment by paperdave** [comment]
oops should not close this via the webkit pr but by the bun pr

**Comment by paperdave** [comment]
Landed in #12023

**Comment by aralroca** [comment]
Cool, thanks @paperdave
